sql/wsrep_encryption.cc

```diff
@@ -0,0 +1,136 @@
+/* Copyright (C) 2019 Codership Oy <info@codership.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */\
+
+#include "my_global.h"
+#include "mysqld.h" // encrypt_binlog
+#include "wsrep_encryption.h"
+#include "wsrep_server_state.h"
+
+static const unsigned int key_version_preamble_size = 8;
+
+/**
+ * Serialize key_version and key into buffer
+ * 
+ * @param input Container for serialized key
+ * @param key Pointer to encryption key
+ * @param size Size of key
+ * @param version Key version
+ */
+static void wsrep_key_serialize(std::vector<unsigned char>& input, const void* key,
+                                size_t size, unsigned int version)
+{
+  input.resize(key_version_preamble_size + size);
+  memcpy(input.data(), (void *)&version, sizeof(version));
```

Comment:
If I correctly understand this is not be/le agnostic, but not sure if that matters.